operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/SSABasedGenericKubernetesResourceMatcher.java

```diff
@@ -104,28 +113,39 @@ public boolean matches(R actual, R desired, Context<?> context) {
   /**
    * Correct for known issue with SSA
    */
-  @SuppressWarnings("unchecked")
   private void sanitizeState(R actual, R desired, Map<String, Object> actualMap) {
-    if (desired instanceof StatefulSet) {
-      StatefulSet desiredStatefulSet = (StatefulSet) desired;
-      StatefulSet actualStatefulSet = (StatefulSet) actual;
-      int claims = desiredStatefulSet.getSpec().getVolumeClaimTemplates().size();
-      if (claims == actualStatefulSet.getSpec().getVolumeClaimTemplates().size()) {
+    if (actual instanceof StatefulSet) {
```

Comment:
Had to adapt this from `if (desired instanceof StatefulSet desiredStatefulSet)` to the good old cast style. This should be the main difference between the original fix and the backport.